### PR TITLE
elscreen-buffer-list.el

### PR DESCRIPTION
**Pull request by abend**
Hi.  Thanks for packaging up and organizing elscreen, and for fixing the bugs with recent emacsen.  I also use elscreen-buffer-list.el, as referenced on the EmacsWiki page for elscreen, so I added it to my repo.  (I'm not the original author.)  If you think it'd be appropriate feel free to add to yours.

### CODE REVIEW

**Comment by zenspider** [comment]
elscreen.el is GPL2 and elscreen-buffer-list.el takes code from elscreen.el (see lines 6-26 in this PR) so that applies GPL2 to it as well.

File: elscreen-buffer-list.el [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      (setq new-list (buffer-list))
+      (while (not (eq (car new-list) firstbuf))
+        (bury-buffer (car new-list))
+        (setq new-list (cdr new-list))))))
+
+
+(defun elscreen-get-buffer-list (screen)
+  "Return buffer-list of SCREEN."
+  (let ((screen-property (elscreen-get-screen-property screen)))
+    (get-alist 'buffer-list screen-property)))
```
Comment (zenspider) [inline elscreen-buffer-list.el]: This line doesn't work in emacs 24.4 w/ the current version of elscreen. Older versions of elscreen ship with:

```elisp
(defun get-alist (key alist)
  (cdr (assoc key alist)))
```

**Comment by DarwinAwardWinner** [comment]
Wow, that is some *old* code that I've long since forgotten about. As noted, lines 6-26 are automatically under GPL2 or higher since they come from [escreen](http://www.splode.com/~friedman/software/emacs-lisp/src/escreen.el), and I give permission for the rest of the code to be licensed under the same terms.

**Comment by knu** [comment]
@DarwinAwardWinner Great to hear from you!  Any chance you could add some license terms to the file on EmacsWiki?

**Comment by knu** [comment]
I'm merging this, and if there's an update on EmacsWiki I'll integrate it later.

**Comment by DarwinAwardWinner** [comment]
Ok, done.

**Comment by knu** [comment]
Thanks, merged!

**Pull request merged**